### PR TITLE
[JUnit] simplify assertions

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java [hunk 1]
```diff
@@ -102,11 +101,11 @@ else if(actualValue == null)
                 {
                     if(message == null)
                     {
-                        assertTrue("Expected key:" + key + " does not exist in Actual Map", false);
+                        fail("Expected key:" + key + " does not exist in Actual Map");
```
Comment (abhipsaMisra) [inline deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java:104]: nice, I've never used a `fail` assertion before

File: deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java [hunk 2]
```diff
@@ -119,23 +118,23 @@ else if(expectedValue instanceof Map)
...
+                            fail(message);
                         }
                     }
                 }
                 else if(expectedValue instanceof ArrayList)
                 {
                     if(message == null)
                     {
-                        assertTrue("Map failed on " + key + ": " + actualValue + " != " + expectedValue, actualValue.toString().equals(expectedValue.toString()));
+                        Assert.assertEquals("Map failed on " + key + ": " + actualValue + " != " + expectedValue, actualValue.toString(), expectedValue.toString());
```
Comment (abhipsaMisra) [inline deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java:133]: do we need to state the class qualifier here?

Comment (timtay-microsoft) [reply deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java:133]: Nope. Both objects here are strings, so the assertEquals(string, string) overload is called

**Review by abhipsaMisra — APPROVED** [review]

File: deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java (hunk 2 above)
Comment (abhipsaMisra) [reply deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/Helpers.java:133]: No, what I meant is, why do we need to call `Assert.assertEquals` (and not `assertEquals` directly).

**Pull request merged**